File: 26-10-2025_Minimum_Cost_of_ropes.cpp
class Solution {
public:
    long long minCost(vector<long long>& arr) {
        priority_queue<long long, vector<long long>, greater<long long>> pq(arr.begin(), arr.end());
        long long cost = 0;
        while (pq.size() > 1) {
            long long a = pq.top(); pq.pop();
            long long b = pq.top(); pq.pop();
            cost += a + b;
            pq.push(a + b);
        }
        return cost;
    }
};

File: 26-10-2025_Minimum_Cost_of_ropes.java
class Solution {
    long minCost(long arr[]) {
        PriorityQueue<Long> pq = new PriorityQueue<>();
        for (long x : arr) pq.add(x);

        long cost = 0;
        while (pq.size() > 1) {
            long a = pq.poll();
            long b = pq.poll();
            cost += a + b;
            pq.add(a + b);
        }
        return cost;
    }
}
